controllers/storagecluster/noobaa_system_reconciler.go

```diff
@@ -185,13 +185,11 @@ func (r *StorageClusterReconciler) setNooBaaDesiredState(nb *nbv1.NooBaa, sc *oc
 	}
 
 	// Add KMS details to Noobaa spec, only if
-	// cluster-wide encryption or KMS is enabled
-	// ie, sc.Spec.Encryption.ClusterWide or sc.Spec.Encryption.KeyManagementService.Enable is True
-	// and KMS ConfigMap is available
+	// KMS is enabled, along with
+	// ClusterWide encryption OR in a StandAlone Noobaa cluster mode
 	// PS: sc.Spec.Encryption.Enable field is deprecated and added for backward compatibility
-	if sc.Spec.Encryption.Enable ||
-		sc.Spec.Encryption.ClusterWide ||
-		sc.Spec.Encryption.KeyManagementService.Enable {
+	if sc.Spec.Encryption.KeyManagementService.Enable &&
+		(sc.Spec.Encryption.Enable || sc.Spec.Encryption.ClusterWide || r.IsNoobaaStandalone) {
```

Comment:
I forgot what "sc.Spec.Encryption.Enable" means. Is it clusterwide or storageclass encryption?

Reply:
sc.Spec.Encryption.Enable and sc.Spec.Encryption.ClusterWide both means clusterwide and UI is adding both the flags currently...

Reply:
we need to clean up and remove sc.Spec.Encryption.Enable entirely...

Reply:
Yeah, for now atleast, we should add a comment in code. Another PR maybe.

Reply:
we have a comment, but in different file: https://github.com/red-hat-storage/ocs-operator/blob/main/api/v1/storagecluster_types.go#L367

Reply:
and we have a comment here as well (regarding the deprecated flag)

Reply:
I will create a Github issue on ocs and console side to do the clean up...